### ISSUE TITLE
Community: Contribution Opportunities and Standards

### ISSUE THREAD

**Issue by em-ly** [ipfs/community#131]
Objective: Establish a community landing page that shows contributors the opportunities ("help wanted") with Protocol Lab's open source projects, and also present the standardization for contributions to IPFS.

**"help wanted"**
Right now the only way to see all available tasks is this [daunting list here](https://github.com/search?utf8=%E2%9C%93&q=label%3A%22difficulty%3Aeasy%22+user%3AIPFS+is%3Aopen&type=Issues&ref=searchresults), which doesn't emphasize what the most valuable contributions to the project could be. It would be great to have these broken out by project.

**Contribution Guidelines**
We have so many of these pages that it's become daunting and not useful. We want this information to be communicated clearly and I feel like we could use some reform in the way information is presented here. Would love to hear your thoughts! 

Links in question:
[IPFS/Community Contributing Guidelines](https://github.com/ipfs/community/blob/master/contributing.md)
[IPFS/Community Contribution Guidelines cont.](https://github.com/ipfs/community/blob/master/contribution-guidelines.md)
[IPFS Code of Conduct](https://github.com/ipfs/community/blob/master/code-of-conduct.md) (should be it's own page, but wanted to be sure it gets included)
[Go Contributing Guidelines](https://github.com/ipfs/community/blob/master/go-contribution-guidelines.md)

**Comment by Kubuxu** [comment]
It is better to use "issue search" as it provides much cleaner interface: https://github.com/issues?utf8=%E2%9C%93&q=label%3A%22difficulty%3Aeasy%22+user%3AIPFS+is%3Aopen

**Comment by RichardLitt** [comment]
I agree with @Kubuxu that the Issue search for the `help wanted` label is useful. But it is really only great for long-time GitHub users who are familiar with cross-repo issue searching, and who are OK with seeing a lot of issues. This might be scary for new developers, developers who want to easily see the best place to start, and for people unfamiliar with the scope of the project. 

A nice interface -- something like [yourfirstpr](https://yourfirstpr.github.io/) -- might be nice. 

Personally, I'm not sure how to get over this. It might be useful to more actively maintain Help Wanted badges, and to link them from each project, so that people can enter from any repo where we want more developers and find issues in that repo easily. I think that might up the visibility, probably more than a separate page that could go out of date or be hard to find (not that it wouldn't be great!)

**Comment by em-ly** [comment]
@RichardLitt let's touch base on this/community dashboard and prioritize it for before team week - if not this week then next.

**Comment by RichardLitt** [comment]
I'm thinking over this again, and I think there's a good deal of discussion that needs to happen before this becomes a priority. Right now, it is a bit vague. 

> "help wanted"
> Right now the only way to see all available tasks is this daunting list here, which doesn't emphasize what the most valuable contributions to the project could be. It would be great to have these broken out by project.

That list _is_ broken out by project. You can search individual repositories. 

If people are interested in more valuable contributions, they can attend the Monday sprints or read the Captain Logs. We could prioritize them - I think this would go a long way towards better guiding contributions than upkeep of another page. 

> Contribution Guidelines
> We have so many of these pages that it's become daunting and not useful. We want this information to be communicated clearly and I feel like we could use some reform in the way information is presented here. Would love to hear your thoughts!

I disagree with this. They have different functions. We could potentially collapse some of them. What's good to remember is that most devs (I think) view contribution guidelines as references - they read through them once, and then refer to it as needed when someone asks them to reformat a PR or something.

**Comment by em-ly** [comment]
The goal of this is to have one location for open source contributors to see what the needs are for different projects. Can they find issues with help wanted per project now? Yep, but we want to make it as easy and convenient (and not overwhelming) as possible for anyone who wants to get tapped into our open source projects as possible. Not everyone has the flexibility to join the sprints (while that would be ideal), and people shouldn't have to wait a week for the next sprint to see what some of the needs are for each of the projects.

I say "some" of the needs because I don't intend for this to be a master list of all help wanted needs, but rather showcasing a few opportunities of how people can contribute.

This is also where we'll be able to have a "top contributor" running list, which has been requested by a few others.

Re: Contribution guidelines
Let's consolidate and standardize where possible. Why do we have two IPFS contributor guideline pages? Why not just one?

**Comment by haadcode** [comment]
A low hanging fruit to improve our community docs here is: a_dd links to the main README in this repo for the guidelines_. Just had a conversation re. JS guidelines and came to look for help. Didn't see them on the front page and was confused until I realized they're files in the repo.

@RichardLitt what do you think?

**Comment by RichardLitt** [comment]
@haadcode SGMT. Can someone review [this PR](https://github.com/ipfs/community/pull/136) before I make another PR to add those?

**Comment by diasdavid** [comment]
Good reading on how we approach this today at https://github.com/ipfs/js-ipfs/blob/master/MGMT.md#core-dev-team-work-tracking--managment. Closing this issue 🕸🕷
